### DISCUSSION TITLE
App-only access (With client secret, without user login) [Sharepoint/OneDrive Business]

### DISCUSSION THREAD

**Discussion by ghost** [abraunegg/onedrive#2548] [Q&A] [answered]
Hello there,

From [Microsoft graph api docs](https://learn.microsoft.com/en-us/graph/auth/auth-concepts) there are two ways to authenticate: delegated access (with a user) and app only access (with a client secret).

From the user documentation and config file documentation, I only found instructions for user login ("After installing the application, you must authorise the application with your OneDrive account").

Is there a way to use app only access or can I open a feature request?

Thanks :)

**Comment by abraunegg** [comment, marked as answer]
@siraponte 
> Is there a way to use app only access

No there is not - sorry. It is currently user-auth access only.

> ,,, or can I open a feature request?

If you are looking for this capability, then yes, you will need to raise a feature request, however this request will not be prioritised. The work focus at the moment is in finishing the re-write for v2.5.x & then fixing issues that arise out of that during beta testing & release.

**Reply by ghost** [reply]
Ok, Thanks :)
